### PR TITLE
V1 and V2 split allocators behind feature flag

### LINKED ISSUE #2561: V2: Support payout to split allocator in UI
## V2 reconfiguration + edit payouts
- [x] **Add payout** modal: add an `allocator` dropdown to payouts modal. Initially will only have one option: the V2SplitAllocator https://github.com/jbx-protocol/juice-v3-migration/tree/main/contracts (note this contract isnt deployed yet)
- [x] **Review and Deploy** section: update payout line item to indicate an allocator is being used.

Screenshot 2022-11-25 at 10 46 31 AM

## V2 project page
- [ ] **funding distribution** card: update payout line item to indicate an allocator is being used.

Screenshot 2022-11-25 at 10 40 24 AM

### LINKED ISSUE #2630: V1: Support payout to split allocator in UI
## V1 reconfiguration + edit payouts
- [ ] **Add payout** modal: add an `allocator` dropdown to payouts modal. Initially will only have one option: the V1SplitAllocator https://github.com/jbx-protocol/juice-v3-migration/tree/main/contracts (note this contract isnt deployed yet)
- [ ] **Review and Deploy** section: update payout line item to indicate an allocator is being used.

Screenshot 2022-11-25 at 10 46 31 AM

## V1 project page
- [ ] **funding distribution** card: update payout line item to indicate an allocator is being used.

Screenshot 2022-11-25 at 10 40 24 AM

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Closes #2561 
Closes #2630

- Hidden behind `splitAllocators` feature flag (enable with `localStorage.setItem('splitAllocators_mainnet', true)`)

## Screenshots or screen recordings

Video demonstrates an `allocator` value passed into the **V2** split tx:

https://user-images.githubusercontent.com/96150256/206342189-06f8c8ba-e9cc-4161-9296-342a1573edcd.mp4

Screen Shot 2022-12-08 at 1 02 58 pm

Screen Shot 2022-12-08 at 2 44 55 pm

**V1:**

https://user-images.githubusercontent.com/96150256/206902273-68504a29-d305-40a0-87be-9d541022e3df.mp4

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by johnnyd-eth** [comment]
All done here @tomquirk. Only question is: not sure if we want to show the version badge for `V3` as well? Or just `V2` (and `V1` when it's done)

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { t, Trans } from '@lingui/macro'
+import * as constants from '@ethersproject/constants'
+import { Form, InputNumber, Radio } from 'antd'
+import { stringIsDigit } from 'utils/math'
+import { featureFlagEnabled } from 'utils/featureFlags'
+import { FEATURE_FLAGS } from 'constants/featureFlags'
+
+export const V2_ALLOCATOR_ADDRESS = 'TODO_V2_ALLOCATOR_ADDRESS'
+const V3_ALLOCATOR_ADDRESS = constants.AddressZero
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:9]: **suggestion**: perhaps `constants/contracts/mainnet` is a better place for these, WDYT?

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      {allocatorsEnabled ? (
+        <Form.Item
+          label={t`Project treasury version`}
+          name="allocator"
+          initialValue={V3_ALLOCATOR_ADDRESS}
+        >
+          <Radio.Group>
+            <Radio value={V2_ALLOCATOR_ADDRESS}>
+              <Trans>V2</Trans>
+            </Radio>
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:39]: **issue** When V2 is selected, there is no allocator set. Only need to set allocator when V3 is selected.

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    // TODO: check if projectId exists
+    return Promise.resolve()
+  }
+
+  const allocatorsEnabled = featureFlagEnabled(FEATURE_FLAGS.SPLIT_ALLOCATORS)
+
+  return (
+    <>
+      {allocatorsEnabled ? (
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:30]: **issue** this should only be shown if CV == 2.

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        <Form.Item
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:31]: **suggestion** WDYT about having this in its own component?

**Review by tomquirk — CHANGES_REQUESTED** [review]
@johnnyd-eth this is awesome thank you for working on it. Left a few comments.

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:31]: Good call

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:30]: Got it

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:39]: GOtcha

File: src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+}) {
+  // need state here (as well as form value in parent component)
+  // to handle the changing of the V1/V3 project input field
+  const [allocator, setAllocator] = useState<string | undefined>(
+    initialAllocator,
+  )
+
+  const allocatorsEnabled = featureFlagEnabled(FEATURE_FLAGS.SPLIT_ALLOCATORS)
+
+  const _V1ProjectHandleFormItem = (
```
Comment (tomquirk) [inline src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx:30]: **nitpick** no need for `_` i think

```suggestion
  const V1ProjectHandleFormItem = (
```

File: src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    return _V1ProjectHandleFormItem
+  }
+  return (
+    <>
+      <Form.Item label={t`Project treasury version`} name={'allocator'}>
+        <Radio.Group
+          value={allocator}
+          onChange={e => setAllocator(e.target.value)}
+        >
+          <Radio value={NULL_ALLOCATOR_ADDRESS}>
```
Comment (tomquirk) [inline src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx:54]: **question** does it default to V1 selected? If not can we do that?

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import { t } from '@lingui/macro'
+import { Form, InputNumber } from 'antd'
+import { stringIsDigit } from 'utils/math'
+import { AllocatorFormItem } from './AllocatorFormItem'
+
+export function V2ProjectPayoutFormItem({
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/V2ProjectPayoutFormItem.tsx:6]: **suggestion** maybe we rename this V2V3 while we're here.

**Review by tomquirk — APPROVED** [review]
@johnnyd-eth this is fantastic, awesome work. Please keep in touch with contract crew to get these addresses and test fully on goerli before enabling feature flag.

File: src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx (hunk 6 above)
Comment (johnnyd-eth) [reply src/components/v1/shared/formItems/ProjectPayoutFormItem.tsx:54]: Yep, it does

**Pull request merged**